java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java

```diff
@@ -149,10 +150,16 @@ public HttpEventResponse handle(HttpEventRequest request) {
 
         boolean tokenizedURLReversed = ObjectUtils.notEqual(requestedTargetUrl, clearTargetUrl);
 
-        URL targetUrl = new URL(clearTargetUrl);
+        // Using original URL to check sanitized rules, as they should match the original URL. It could contain tokenized components.
```

Comment:
They don't though. The rules restrict calls to REQUIRE pseudonymized params; otherwise it's a logical hole - eg I could call `users/123` and receive the pseudonym for that user in the response, which would allow me to re-identify the data.



![SCR-20240628-ijxs](https://github.com/Worklytics/psoxy/assets/102831/a3e97684-a161-47d4-bbd9-23f07fa08129)